OpenSim/Common/Reporter.h

```diff
@@ -0,0 +1,252 @@
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2016 Stanford University and the Authors                *
+ * Author(s): Ajay Seth                                        *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+// INCLUDE
+#include <OpenSim/Common/Component.h>
+#include <OpenSim/Common/TimeSeriesTable.h>
+
+namespace OpenSim {
+
+/**
+ * This abstract class represents a Reporter that generates a report from 
+ * values generated by Components in a Model during a simulation. 
+ *
+ * The what, how and where valuess are reported (e.g. to the console, 
+ * DataTable, device, etc...) are the purview of concrete Reporters.
+ *
+ * @author Ajay Seth
+ */
+class OSIMCOMMON_API AbstractReporter : public Component {
+OpenSim_DECLARE_ABSTRACT_OBJECT(AbstractReporter, Component);
+public:
+//==============================================================================
+// PROPERTIES
+//==============================================================================
+    OpenSim_DECLARE_PROPERTY(is_disabled, bool,
+        "Flag indicating whether the Reporter is disabled or not. Disabled means"
+        " that the Reporter does not record in subsequent Report realizations.");
+
+    OpenSim_DECLARE_PROPERTY(report_time_interval, double,
+        "The recording time interval (s). If interval < 0 or NaN defaults to"
+        "every valid integration time step.");
+
+//=============================================================================
+// PUBLIC METHODS
+//=============================================================================
+    /** Report values given the state and top-level Component (e.g. Model) */
+    void report(const SimTK::State& s) const;
```

Comment:
It's not clear to me that this method is necessary. But perhaps it's useful if someone is writing their own homebrewed solver in Python or MATLAB.